### PR TITLE
Chore: Adds eslint

### PR DESCRIPTION
**Pull request by hugohaggmark**
This PR adds eslint config and extends the Grafana eslint config. There were a 18 warnings and 1 error all related to React Hooks that I've fixed. I'll leave comments in the files where I'm not sure why we're missing dependencies or other issues.

### CODE REVIEW

File: src/QueryEditor.tsx [hunk 1]
```diff
@@ -41,24 +42,25 @@ export const QueryEditor: React.FC<Props> = props => {
       props.onRunQuery();
     }
   }, []);
```
Comment (hugohaggmark) [inline src/QueryEditor.tsx:44]: I guessed that we only want to check this once on component mount?

File: src/editor/components/field/QueryEditorFieldSection.tsx [hunk 2]
```diff
@@ -40,11 +32,18 @@ export const QueryEditorFieldSection = (config: FieldSectionConfiguration): Reac
...
     );
   };
 };
+
+const onChange = (propsOnChange: (value: QueryEditorExpression) => void) => (property: QueryEditorProperty) => {
+  propsOnChange({
+    type: QueryEditorExpressionType.Property,
+    property,
+  } as QueryEditorPropertyExpression);
+};
```
Comment (hugohaggmark) [inline src/editor/components/field/QueryEditorFieldSection.tsx:49]: The only way I figured that we could solve the eslint error with was to extract a proper function

File: src/editor/components/groupBy/QueryEditorGroupBy.tsx [hunk 3]
```diff
@@ -47,9 +49,10 @@ export const QueryEditorGroupBy: React.FC<Props> = props => {
         interval,
       };
 
-      props.onChange(payload);
+      onChange(payload); // adding onChange to dependency array below causes maximum call depth error
```
Comment (hugohaggmark) [inline src/editor/components/groupBy/QueryEditorGroupBy.tsx:52]: I wrote a comment about this one. If I add onChange to dependencies we get a recursion with a maximum call stack depth. Didn't look into this further, but we really should dive deeper into this when we have time.

File: src/editor/components/reduce/QueryEditorReduce.tsx [hunk 4]
```diff
@@ -74,9 +76,10 @@ export const QueryEditorReduce: React.FC<Props> = props => {
         parameters: parameters,
       };
 
-      props.onChange(payload);
+      onChange(payload); // adding onChange to dependency array below causes maximum call depth error
```
Comment (hugohaggmark) [inline src/editor/components/reduce/QueryEditorReduce.tsx:79]: I wrote a comment about this one. If I add onChange to dependencies we get a recursion with a maximum call stack depth. Didn't look into this further, but we really should dive deeper into this when we have time.

File: src/QueryEditor.tsx (hunk 1 above)
Comment (mckn) [reply src/QueryEditor.tsx:44]: yes 👍

File: src/editor/components/field/QueryEditorFieldSection.tsx (hunk 2 above)
Comment (mckn) [reply src/editor/components/field/QueryEditorFieldSection.tsx:49]: Feels like a good solution! The other one is only an optimization that can be skipped for now.

File: src/editor/components/groupBy/QueryEditorGroupBy.tsx (hunk 3 above)
Comment (mckn) [reply src/editor/components/groupBy/QueryEditorGroupBy.tsx:52]: hmm, it looks like we are doing something really bad here and it should probably be changed. Lets add this one to the list. I'm not really following what we are trying to solve so need to have a look at that before I can suggest a solution.

**Review by mckn — APPROVED** [review]
It looks really good!

We need to have a look at how we manage state with hooks in the two examples that you added comments for below.

**Pull request merged**